Improved formatting of examples; Other misc edits (#7782)
yugabyte/yugabyte-db: docs/content/latest/explore/ysql-language-features/queries.md

### Case Sensitivity

> YSQL converts identifiers to lowercase unless they are enclosed in quotation marks. That is, YSQL is not case-sensitive for all practical purposes by default. For example, a table called `Employees` would be recognized as the `employees` table and the following query would be executed on the `employees` table without any problems:

Reviewer: ```suggestion
YSQL converts identifiers to lowercase unless they are enclosed in quotation marks. That is, YSQL is case-insensitive for all practical purposes by default. For example, a table called `Employees` would be recognized as the `employees` table and the following query would be executed on the `employees` table without any problems:
```

Author: Changed

```sql

> SELECT name FROM Employees;

Reviewer: ```suggestion
SELECT name FROM employees;
```

Author: It needs to be Employees

If there were duplicate rows in the result set from the preceding example, these rows would have been removed because the `GROUP BY` clause functions like the `DISTINCT` clause in this case.

> If the `employees` table had a column called `amount` that contained the employee pay data, the following example would have shown how to select the total amount that each employee has been paid. The `GROUP BY` clause would have divided the rows in the `employees` table into groups by employee number. For each group, the total amounts would have been calculated using the `SUM()` function:

Reviewer: Too much conditional in here. Can you rework it so it actually _does_ show the things you say it _would have_ shown? And show sample output? (At which point, the previous wording is probably good...)
```suggestion
If the `employees` table had an `amount` column that contained the employee pay data, the following example would have shown how to select the total amount that each employee has been paid. The `GROUP BY` clause would have divided the rows in the `employees` table into groups by employee number. For each group, the total amounts would have been calculated using the `SUM()` function:
```

Author: Reworded slightly, but don't have time for a new example setup right now. Will do this some other time (made a note for myself)

EXISTS (SELECT 1 FROM some_table WHERE condition);
```

The following example applies the `EXISTS` operator to two tables (the  `employees` table and a hypothetical  `salary` table) and shows how to use `EXISTS` on the `employee_no` column in a manner similar to an inner join:

```sql
SELECT name FROM employees
  WHERE EXISTS (SELECT 1 FROM salary
                  WHERE salary.employee_no = employees.employee_no);
```

If the `salary` table existed, the preceding query would have returned no more than one row for each row in the `employees` table, even though there would have been corresponding rows in the `salary` table.

## Recursive Queries and CTEs

> Common Table Expressions (CTEs) allow you to execute recursive, hierarchical, and other type of complex queries in a simplified manner by breaking down these queries into smaller units. A CTE exists only during the query execution and represents a temporary result set that you can reference from another SQL statement.

Reviewer: ```suggestion
Common Table Expressions (CTEs) allow you to simplify recursive, hierarchical, and other types of complex queries by breaking them down into smaller units. A CTE exists only during query execution, and is a temporary result set that you can reference from another SQL statement.
```

Author: Reworded this sentence on purpose (at the expense of elegance)

  manager_no integer,
  department text
);
```

```sql
INSERT INTO employees VALUES 

> (1221, 'John Smith', NULL, NULL),

Reviewer: why delete the leading spaces (here and lots of other places)? they work fine on my machine, and make it easier to read the multi-line statements...

Author: Added spaces here, but I've had so many problems with them that I'd rather keep examples less readable. It could be related to the user's browser. e.g, I use Firefox

(1225, 'Walter Marx', 1222, 'Sales');
```

The following example demonstrates how to retrieve all employees who report to the manager with `manager_no` 1222:

```sql
WITH RECURSIVE reports AS (
  SELECT employee_no, name, manager_no, department 
  FROM employees 
  WHERE employee_no = 1222 
  UNION 
    SELECT 
      emp.employee_no, emp.name, emp.manager_no, emp.department 
    FROM employees emp 
    INNER JOIN reports rep ON rep.employee_no = emp.manager_no
) 
SELECT * FROM reports;
```

In the preceding example, the recursive CTE `reports` defines a non-recursive and a recursive term, with non-recursive term returning the base result set that includes the employee whose `employee_no` is 1222. Then the recursive term retrieves the direct reports of the employee whose `employee_no`  is 1222. This is the result of joining the `employees` table and the `reports` CTE. The first iteration of the recursive term returns two employees with  `employee_no`  1224 and 1225. The recursive term is executed multiple times. The second iteration uses the preceding result set as the input value but does not return any rows because nobody reports to employees with `employee_no` 1224 and 1225.

> When the final result set is returned, it represents the combination of all result sets in iterations generated by the non-recursive  and recursive terms, as demonstrated by the following output:

Reviewer: Make sure this is correct :)

Author: !!
